src/cascadia/TerminalApp/CommandPalette.cpp

```diff
@@ -870,31 +876,40 @@ namespace winrt::TerminalApp::implementation
             _allCommands.Append(filteredCommand);
         }
 
-        _updateFilteredActions();
+        if (Visibility() == Visibility::Visible && _currentMode == CommandPaletteMode::ActionMode)
+        {
+            _updateFilteredActions();
+        }
     }
 
-    void CommandPalette::SetTabs(Collections::IVector<TabBase> const& tabs, const bool clearList)
+    // Method Description:
+    // - Replaces a list of filtered commands in the target collection with new
+    //   commands based on the tabs in the source collection.
+    // Although the source observable we still don't register on it,
+    // so the palette user will need to reset the binding manually every time
+    // the source collection changes
+    // Arguments:
+    // - source: the tabs to use for creation filtered commands
+    // - target: the collection to store newly created filtered commands
+    // Return Value:
+    // - <none>
+    void CommandPalette::_bindTabs(
+        Windows::Foundation::Collections::IObservableVector<winrt::TerminalApp::TabBase> const& source,
+        Windows::Foundation::Collections::IVector<winrt::TerminalApp::FilteredCommand> const& target)
     {
-        _tabActions.Clear();
-        for (const auto& tab : tabs)
+        target.Clear();
+        for (const auto& tab : source)
         {
             auto tabPaletteItem{ winrt::make<winrt::TerminalApp::implementation::TabPaletteItem>(tab) };
             auto filteredCommand{ winrt::make<FilteredCommand>(tabPaletteItem) };
-            _tabActions.Append(filteredCommand);
+            target.Append(filteredCommand);
         }
+    }
 
-        // The smooth remove/add animations that happen during
-        // UpdateFilteredActions don't work very well with changing the tab
-        // order, because of the sheer amount of remove/adds. So, let's just
-        // clear & rebuild the list when we change the set of tabs.
-        //
-        // Some callers might actually want smooth updating, like when the list
-        // of tabs changes.
```

Comment:
Does this mean that now, if a tab is removed which the switcher is open, that the entire list will re-animate into existence, rather than the smooth update it does now?

Reply:
Now if the tab is removed our ephemeral palette gets closed immediately :blush:

Reply:
I'm okay with this

Reply:
I believe that's because of the earlier change to dispatch bindings from TerminalPage::PreviewKeyDown, right? That makes sense to me.

But... does this also hold if a user (trying to break terminal :P) issues a `sleep 5; exit` and then opens up the palette and waits for it to blow up?

Reply:
@DHowett - what a nasty user :blush: I believe that the problem you describe exists in the original code as well.

From my understanding,  If the user triggers delayed sleep and then starts switching, what will happen right now is that we will still have TabPaletteItem populated in the switcher, but switching to it will have no action (as weak ref is empty). Not perfect, not terrible (or however they said this in Chernobyl). To resolve this we need to have the fully fledged binding that I planned for the next PR (where we do follow the changes in _tabs, _mruTabs that are observable collections).

Reply:
@DHowett - ignore the previous comment. What happens now, if the tab is closed by `sleep 5; exit` we actually dismiss the switcher. This was introduced as a part of the ephemeral palette work (every change in tabs collection closes the palette) to ensure consistency. If the user tries to break terminal this way, the only impact that the palette gets closed.